pkg/runtime/k0s/reset.go

```diff
@@ -0,0 +1,81 @@
+// Copyright © 2022 Alibaba Group Holding Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k0s
+
+import (
+	"context"
+	"fmt"
+	"net"
+
+	"golang.org/x/sync/errgroup"
+)
+
+func (k *Runtime) reset() error {
+	if err := k.resetNodes(k.cluster.GetNodeIPList()); err != nil {
+		return err
+	}
+	if err := k.resetMasters(k.cluster.GetMasterIPList()); err != nil {
+		return err
+	}
+
+	return k.DeleteRegistry()
+}
+
+func (k *Runtime) resetNodes(nodes []net.IP) error {
+	eg, _ := errgroup.WithContext(context.Background())
+	for _, node := range nodes {
+		node := node
+		eg.Go(func() error {
+			if err := k.resetNode(node); err != nil {
+				return fmt.Errorf("failed to reset node %s: %v", node, err)
+			}
+			return nil
+		})
+	}
+	return eg.Wait()
+}
+
+func (k *Runtime) resetMasters(nodes []net.IP) error {
```

Comment:
Do we need to use a `eg, _ := errgroup.WithContext(context.Background())` in `resetMasters` as well, just like what you did in `resetNodes`? @starComingup

Reply:
I think we can do what you advice, then I will verify it.